s3/s3_transport/unit_tests/src/test_s3_transport.cpp

```diff
@@ -0,0 +1,995 @@
+#include "catch.hpp"
+
+#include "s3_transport.hpp"
+#include "s3_transport_util.hpp"
+#include "s3_multipart_shared_data.hpp"
+#include <filesystem/filesystem.hpp>
+#include <dstream.hpp>
+#include <mutex>
+#include <condition_variable>
+//#include <jansson.h>
+#include <fstream>
+#include <thread>
+#include <chrono>
+#include <sys/wait.h>
+#include <stdexcept>
+#include <cstdio>
+
+#include <string_view>
+
+using odstream            = irods::experimental::io::odstream;
+using idstream            = irods::experimental::io::idstream;
+using dstream             = irods::experimental::io::dstream;
+using s3_transport        = irods::experimental::io::s3_transport::s3_transport<char>;
+using s3_transport_config = irods::experimental::io::s3_transport::config;
+
+namespace fs = irods::experimental::filesystem;
+namespace io = irods::experimental::io;
+
+std::string keyfile = "/projects/irods/vsphere-testing/externals/amazon_web_services-CI.keypair";
+std::string hostname = "s3.amazonaws.com";
+std::string bucket_name = "justinkylejames1";
```

Comment:
less justin?

Reply:
Added in code to generate the bucket name using the current epoc time + milliseconds with a header.  Create the bucket on setup and remove on cleanup.  

Removed the --bucket-name option from the unit test program.

Ran unit tests and verified they pass and that the buckets are cleaned up after they run.